lib/spack/spack/test/config_requirements.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

Comment:
(minor) Should we use `concretize_requirements.py` (as opposed to `concretize_preferences.py`)?